src/strategies/BeefyStrategy.sol

```diff
@@ -0,0 +1,148 @@
+        zapper.add_liquidity(address(curvePool), amounts, minLpToken);
+
+        // deposit lpToken in beefy
+        beefy.depositAll();
+    }
+
+    function _divest(uint256 assets) internal override returns (uint256) {
+        uint256 amount = divestFromBeefy(assets, defaultSlippageBps);
+        uint256 amountToSend = Math.min(assets, amount);
+        asset.safeTransfer(address(vault), amountToSend);
+        return amountToSend;
+    }
+
+    function divestFromBeefy(uint256 assets, uint256 slippageBps) internal returns (uint256) {
+        if (asset.balanceOf(address(this)) > assets) {
+            return assets;
+        }
+
+        uint256 requiredAssets = assets - asset.balanceOf(address(this));
+
+        uint256[4] memory amounts = [uint256(0), 0, 0, 0];
+        amounts[uint256(uint128(assetIndex))] = requiredAssets;
+
+        uint256 requiredLpToken = zapper.calc_token_amount(address(curvePool), amounts, false);
+        // need to withdraw more due to slippage
+        uint256 lpTokenToWithdraw = requiredLpToken.mulDivDown(MAX_BPS, MAX_BPS - slippageBps);
+
+        withdrawLPTokenFromBeefy(lpTokenToWithdraw);
+
+        // remove liquidity from curve
+        // @dev withdraw full amount, so that no curve token left idle.
+        removeLiquidityFromCurve(slippageBps);
+
+        // only withdrawing required assets
+        return asset.balanceOf(address(this));
+    }
+
+    function removeLiquidityFromCurve(uint256 slippageBps) internal {
+        uint256 withdrawableAssets =
+            zapper.calc_withdraw_one_coin(address(curvePool), curvePool.balanceOf(address(this)), assetIndex);
+
+        uint256 minAssets = withdrawableAssets.mulDivDown(MAX_BPS - slippageBps, MAX_BPS);
+
+        zapper.remove_liquidity_one_coin(address(curvePool), curvePool.balanceOf(address(this)), assetIndex, minAssets);
+    }
+
+    function withdrawLPTokenFromBeefy(uint256 lpTokenAmount) internal {
+        // shares to withdraw
+        uint256 beefyShareToWithdraw = lpTokenAmount.divWadUp(beefy.getPricePerFullShare());
+
+        // check for min shares
+        beefyShareToWithdraw = Math.min(beefy.balanceOf(address(this)), beefyShareToWithdraw);
+
+        beefy.withdraw(beefyShareToWithdraw);
+    }
+
+    function totalLockedValue() external view override returns (uint256) {
+        uint256 lpTokenAmount =
+            beefy.balanceOf(address(this)).mulWadDown(beefy.getPricePerFullShare()) + curvePool.balanceOf(address(this));
+
+        if (lpTokenAmount < 3) {
+            // @dev providing values less than 3 incurs evn crash, not returning zero zero
+            // on a 18 decimal place 0,1,2 means nothing
+            return asset.balanceOf(address(this));
+        }
+        return zapper.calc_withdraw_one_coin(address(curvePool), lpTokenAmount, assetIndex)
+            + asset.balanceOf(address(this));
+    }
+
+    function setDefaultSlippageBps(uint256 slippageBps) external onlyGovernance {
```

Comment:
I think you can just let the strategist (STRATEGIST_ROLE) do this. This sort of detail is too small for governance to handle.